### PR TITLE
Creating CheckerGames. also methods to validate if a move is a valid …

### PR DESCRIPTION
**Pull request by maximussage1**
…move.

### CODE REVIEW

**Comment by maximussage1** [comment]
Please don't merge, just review.

File: src/L8/game/checkers/checkers_constans.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
+
+
+class CheckerColors(Enum):
```
Comment (charlieitesm) [inline src/L8/game/checkers/checkers_constans.py:4]: In English Checkers, all pieces must only use the Black squares, nobody is allowed to jump horizontally or vertically, only diagonally (even when capturing pieces).
Knowing this, we can validate, not based on the color on the board, but the direction of the movement. So I think we could do without this Enum.

File: src/L8/game/checkers/checkers_game.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class CheckersGame(Game, ABC):
+
+    def __init__(self, board: Board, players: list):
+        super().__init__(board, players)
+
+    def is_valid_move(self, move: dict, player: Player) -> bool:
+        r1, c1, r2, c2, is_jump, player_color = move[MOVE]
+        move = CheckersMove(r1, c1, r2, c2, is_jump)
```
Comment (charlieitesm) [inline src/L8/game/checkers/checkers_game.py]: How are you reading the origin of the piece that is to be moved? I believe that the current engine is not supporting reading the origin. I was thinking that we might want to adapt some of the older methods in the engine for that purpose.

File: src/L8/game/checkers/checkers_game.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+from L8.player.player import Player
+
+
+class CheckersGame(Game, ABC):
+
+    def __init__(self, board: Board, players: list):
+        super().__init__(board, players)
+
+    def is_valid_move(self, move: dict, player: Player) -> bool:
+        r1, c1, r2, c2, is_jump, player_color = move[MOVE]
```
Comment (charlieitesm) [inline src/L8/game/checkers/checkers_game.py]: The color of the player can be determined by the GameToken he is holding, how are you reading the player_color?

File: src/L8/game/checkers/checkers_utils.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    @staticmethod
+    def is_valid_move(move: CheckersMove, player_color: str, board: CheckersBoard) -> bool:
+        """
+        Checks if move is a legal move by obtaining a list of legal moves and checking if the move is in the list.
+        :param move:
+        :param player_color:
+        :param board:
+        :return: True if it is legal, false otherwise
+        """
+        legal_moves = CheckerGameUtil.get_valid_moves_for_player(board, player_color)
```
Comment (charlieitesm) [inline src/L8/game/checkers/checkers_utils.py:23]: What's the cost of calculating the whole solution space (even for moves that the player is not currently making)? Wouldn't it be better if we calculate if the current move is valid?

File: src/L8/game/checkers/checkers_game.py [hunk 5]
```diff
@@ -11,15 +11,27 @@
...
     def is_valid_move(self, move: dict, player: Player) -> bool:
-        r1, c1, r2, c2, is_jump, player_color = move[MOVE]
-        move = CheckersMove(r1, c1, r2, c2, is_jump)
+        r1, c1, r2, c2 = move[MOVE]
+        if player.game_token is "kw":
+            player_color = "w"
+        elif player.game_token is "kb":
+            player_color = "b"
+        else:
+            player_color = player.game_token
```
Comment (charlieitesm) [inline src/L8/game/checkers/checkers_game.py:23]: Here you are storing player_color with a GameToken object, but on lines 19 and 21 you are storing a str in the same variable.

I would recommend that on lines 19 and 21 you get the actual singleton GameToken object. I believe there is a util method that can fetch you a GameToken instance given a str representation of said token, if there is not a method, we could add it.

By not keeping the typing consistent, you might run into issues if you are using if's to compare values, after all, a str object will not evaluate to True against a GameToken object (unless you transform either to the other value).

**Review by charlieitesm — CHANGES_REQUESTED** [review]
Just a minor thing there with typing for the player's game token, otherwise it looks pretty solid!

**Comment by charlieitesm** [comment]
I merged these changes @maximussage1 so that we can continue iterating on it. We can address the changes on a following PR, in the meantime I'll move continue adding the Factory and the parameters

**Pull request merged**